### PR TITLE
Use github forms for issue templates

### PR DESCRIPTION
**Pull request by thockin**
You can see the results here: https://github.com/thockin/kubernetes/issues/new/choose

Sadly it doesn't seem to work for PRs.

/kind cleanup

```release-note
NONE
```

### CODE REVIEW

**Comment by aojea** [comment]
lgtm
/assign @mrbobbytables

**Comment by mrbobbytables** [comment]
Ah...sorry, this slipped through the cracks for me. Will review shortly 👍

File: .github/ISSUE_TEMPLATE/enhancement.yaml [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+body:
+  - type: textarea
+    id: feature
+    attributes:
+      label: What would you like to be added?
+      placeholder: |
+        NOTE: Feature requests are unlikely to make progress as an issue.  Instead, please consider engaging with SIGs on slack and mailing lists.
+
+        A proposal that works through the design along with the implications of the change can be opened as a KEP.
+        See https://git.k8s.io/enhancements/keps#kubernetes-enhancement-proposals-keps
```
Comment (mrbobbytables) [inline .github/ISSUE_TEMPLATE/enhancement.yaml:13]: Not necessarily a blocker, but no one can copy paste from placeholder text. We may want to either add a [description](https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/syntax-for-githubs-form-schema#attributes-1) with the link, similar to the [Netlify site request form](https://github.com/kubernetes/org/issues/new?assignees=&labels=area%2Fgithub-integration&template=site-request.yml&title=REQUEST%3A+Netlify+site+for+%3Crepo%3E) or create a shortlink like go.k8s.io/keps that could be used as an easy to type shortcut that could be used in the placeholder text.

File: .github/ISSUE_TEMPLATE/flaking-test.yaml [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+  - type: textarea
+    id: jobs
+    attributes:
+      label: Which jobs are flaking?
+      placeholder: |
+        Please only use this template for submitting reports about flaky tests or jobs (pass or fail with no underlying change in code) in Kubernetes CI.
+
+        Links to go.k8s.io/triage and/or links to specific failures in spyglass are appreciated.
+
+        Please see the deflaking doc (https://github.com/kubernetes/community/blob/master/contributors/devel/sig-testing/flaky-tests.md) for more guidance.
```
Comment (mrbobbytables) [inline .github/ISSUE_TEMPLATE/flaking-test.yaml:14]: Similar to the enhancements tracking issue form, we'd probably need to link out in a description or create a shortlink for the long URL.

**Review by mrbobbytables — CHANGES_REQUESTED** [review]
Thanks! This has actually been on my todo for a bit, but fell into the realm of E_TOO_MANY_THINGS. 

I have a few small suggestions, and after that it should be good to go :)

File: .github/ISSUE_TEMPLATE/failing-test.yaml [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+
+  - type: textarea
+    id: reason
+    attributes:
+      label: Reason for failure (if possible)
+
+  - type: textarea
+    id: additional
+    attributes:
+      label: Anything else we need to know?
```
Comment (RinkiyaKeDad) [inline .github/ISSUE_TEMPLATE/failing-test.yaml:41]:
```suggestion
      label: Anything else we need to know?
      
  - type: textarea
    id: sig
    attributes: 
      label: Relevant SIG
      placeholder: |
        You can identify the SIG from the "prowjob_config_url" on the testgrid dashboard for a test.
    validations: 
      required: true
```

File: .github/ISSUE_TEMPLATE/flaking-test.yaml [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+
+  - type: textarea
+    id: reason
+    attributes:
+      label: Reason for failure (if possible)
+
+  - type: textarea
+    id: additional
+    attributes:
+      label: Anything else we need to know?
```
Comment (RinkiyaKeDad) [inline .github/ISSUE_TEMPLATE/flaking-test.yaml:45]:
```suggestion
      label: Anything else we need to know?
      
  - type: textarea
    id: sig
    attributes: 
      label: Relevant SIG
      placeholder: |
        You can identify the SIG from the "prowjob_config_url" on the testgrid dashboard for a test.
    validations: 
      required: true
```

**Review by RinkiyaKeDad — COMMENTED** [review]
Suggested changes based on https://github.com/kubernetes/kubernetes/pull/105190 🙂

Also, it won't be possible to use prow commands from the form entries right? So for example, if someone entered `/sig node` in the form, would prow be able to apply the correct label?

File: .github/ISSUE_TEMPLATE/bug-report.yaml [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+    id: kubeVersion
+    attributes:
+      label: Kubernetes version
+      value: |
+        <details>
+        ```console
+        $ kubectl version
+        # paste output here
+        ```
+        </details>
```
[the remark this answers is not in the document]
Comment (thockin) [reply .github/ISSUE_TEMPLATE/bug-report.yaml]: ACK

File: .github/ISSUE_TEMPLATE/enhancement.yaml (hunk 1 above)
Comment (thockin) [reply .github/ISSUE_TEMPLATE/enhancement.yaml:12]: Will try as description

File: .github/ISSUE_TEMPLATE/failing-test.yaml (hunk 3 above)
Comment (thockin) [reply .github/ISSUE_TEMPLATE/failing-test.yaml:41]: ACK

File: .github/ISSUE_TEMPLATE/flaking-test.yaml (hunk 2 above)
Comment (thockin) [reply .github/ISSUE_TEMPLATE/flaking-test.yaml:12]: Description works, I think.  PTAL

**Review by thockin — COMMENTED** [review]
PTAL

**Comment by mrbobbytables** [comment]
> Also, it won't be possible to use prow commands from the form entries right? So for example, if someone entered /sig node in the form, would prow be able to apply the correct label?

They will come through, when posted its all markdown and prow responds to the content of the message

**Pull request merged**